dtool/Config.cmake

```diff
@@ -0,0 +1,629 @@
+#
+# dtool/Config.cmake
+#
+# This file defines certain configuration variables that are written
+# into the various make scripts.  It is processed by CMake to
+# generate build scripts appropriate to each environment.
+#
+
+include(CMakeDependentOption)
+
+# Define our target platform.
+# The values "UNIX", "WIN32", "MINGW", "MSYS", and "CYGWIN"
+# are automatically provided by CMAKE.  "APPLE" is also provided by
+# CMAKE but may be True on systems that are not OS X.
+if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Android")
+  set(IS_LINUX 1)
+endif()
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  set(IS_OSX 1)
+endif()
+if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+  set(IS_FREEBSD 1)
+endif()
+
+if(CMAKE_VERSION VERSION_GREATER "3.8")
+  get_property(IS_MULTICONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+else()
+  message(WARNING "Multi-configuration builds may not work properly when using
+a CMake < 3.9. Making a guess if this is a multi-config generator.")
+  if(DEFINED CMAKE_CONFIGURATION_TYPES)
+    set(IS_MULTICONFIG ON)
+  else()
+    set(IS_MULTICONFIG OFF)
+  endif()
+endif()
+
+# Define the type of build we are setting up.
+if(IS_MULTICONFIG)
+  set(CMAKE_CONFIGURATION_TYPES Release RelWithDebInfo Debug MinSizeRel Distribution)
+else()
+  # CMAKE_BUILD_TYPE can't just be set using the usual set(CACHE) method since
+  # it's an empty string by default.
+  if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE RelWithDebInfo)
+  endif()
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+    Release RelWithDebInfo Debug MinSizeRel Distribution)
+endif()
+
+# Provide convenient boolean expression based on build type
+if(CMAKE_BUILD_TYPE MATCHES "Debug")
+  set(IS_DEBUG_BUILD True)
+  set(IS_NOT_DEBUG_BUILD False)
```

Comment:
ON/OFF is what I usually use. This is actually old stuff that I thought was phased out (we should not be checking `CMAKE_BUILD_TYPE` at all; it may not even be known at configuration time)

Reply:
In that case, if it's obsolete, it should be removed with prejudice.

Reply:
This one seems unresolved as of yet.